language/move-compiler/src/naming/fake_natives.rs

```diff
@@ -86,37 +87,24 @@ pub fn resolve_builtin(
     function: &FunctionName,
 ) -> Option<fn(Vec<IR::Type>) -> IR::Bytecode_> {
     let sp!(_, ModuleIdent_ { address, module }) = module;
-    let addr_name = match address {
-        Address::Numerical(Some(sp!(_, n_)), _) | Address::NamedUnassigned(sp!(_, n_)) => n_,
+    // Only resolve if either (a) the address is named "std" or (b) its value is 0x1
+    match address {
+        Address::Numerical(Some(sp!(_, n)), sp!(_, a))
+            if n.as_str() == "std" || a.into_inner() == AccountAddress::ONE => {}
+        Address::NamedUnassigned(sp!(_, n)) if n.as_str() == "std" => {}
```

Comment:
well hack upon hack... 

I think the long term best solution is to figure out if we can funnel in the address prefix... but ... yeah... the situation currently is not great.

Reply:
That's actually the cleanest we can get (relative to the language definition). Here at this point we have both the symbolic and the numerical form of the address. We want to support that someone deploys `std` at a different address. And we want to support that users just write `0x1`. Thats what this enables.

The problem that we have on the one hand numeric addresses and on the other symbolic ones which can be aliases for the former is inherent to the address alias system. Only a change in the language can fix this (e.g. disallowing numeric addresses all together.)